modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -475,8 +482,14 @@ protected TableHeaderRow getTableHeaderRow() {
         return tableHeaderRow;
     }
 
-    void setTableHeaderRow(TableHeaderRow thr) {
+   void setTableHeaderRow(TableHeaderRow thr) {
+        if (tableHeaderRow != null) {
+            tableHeaderRow.cornerPadding.removeListener(cornerPaddingChangeListener);
+        }
         tableHeaderRow = thr;
+        if (tableHeaderRow != null) {
+            tableHeaderRow.cornerPadding.addListener(cornerPaddingChangeListener);
+        }
```

Comment:
question: would it make sense to go through changeListenerHandler?

doing so will create a weak listener intermediary, but might make the code added to dispose() unnecessary.